File: src/main/java/com/dyrosoft/kvbparser/api/ApiProvider.java
package com.dyrosoft.kvbparser.api;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava.RxJavaCallAdapterFactory;

public final class ApiProvider {

    private static class HeaderInterceptor implements Interceptor {

        private static final String HEADER_USER_AGENT =
                "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/34.0.1847.137 Safari/537.36";

        @Override
        public Response intercept(final Chain chain) throws IOException {
            final Request request = chain.request();

            final Request.Builder builder = request.newBuilder();
            builder.addHeader("user-agent", HEADER_USER_AGENT);
            final Request modifiedRequest = builder.build();

            return chain.proceed(modifiedRequest);
        }
    }

    private static final String BASE_URL = "http://www.kvb-koeln.de/";

    private ApiProvider() {
    }

    public static KvbApi getKvbApi() {
        return buildRetrofit().create(KvbApi.class);
    }

    private static Retrofit buildRetrofit() {
        final OkHttpClient client = new OkHttpClient.Builder()
                .addInterceptor(new HeaderInterceptor())
                .build();

        final String endpoint = BASE_URL;
        return new Retrofit.Builder()
                .baseUrl(endpoint)
                .client(client)
                .addCallAdapterFactory(RxJavaCallAdapterFactory.create())
                .build();
    }
}

File: src/main/java/com/dyrosoft/kvbparser/api/ApiProvider.kt
package com.dyrosoft.kvbparser.api

import okhttp3.Interceptor
import okhttp3.OkHttpClient
import okhttp3.Response
import retrofit2.Retrofit
import retrofit2.adapter.rxjava.RxJavaCallAdapterFactory
import java.io.IOException

object ApiProvider {

    private class HeaderInterceptor : Interceptor {

        @Throws(IOException::class)
        override fun intercept(chain: Interceptor.Chain): Response =
                chain.proceed(chain.request()
                                      .newBuilder()
                                      .addHeader("user-agent", HEADER_USER_AGENT)
                                      .build())

        companion object {
            private const val HEADER_USER_AGENT = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/34.0.1847.137 Safari/537.36"
        }
    }

    private const val BASE_URL = "http://www.kvb-koeln.de/"

    val kvbApi: KvbApi
        get() = buildRetrofit().create(KvbApi::class.java)

    private fun buildRetrofit(): Retrofit {
        val client = OkHttpClient.Builder()
                .addInterceptor(HeaderInterceptor())
                .build()

        return Retrofit.Builder()
                .baseUrl(BASE_URL)
                .client(client)
                .addCallAdapterFactory(RxJavaCallAdapterFactory.create())
                .build()
    }
}
